src/attr/_make.py

```diff
@@ -1929,10 +1961,10 @@ def _make_init(
         unique_filename,
     )
 
-    __init__ = locs["__init__"]
-    __init__.__annotations__ = annotations
+    init = locs["__attrs_init__"] if attrs_init else locs["__init__"]
```

Comment:
IIRC correctly, the reason why we used `__init__` as a variable name, was such that `__name__` etc are correct? 🤔  But I don't think it works that way if I think about it…

Reply:
Ya, the retval from this function is assigned to the `_cls_dict`  in `_ClassBuilder`: https://github.com/indigoviolet/attrs/blob/vi-attr-init/src/attr/_make.py#L891, https://github.com/indigoviolet/attrs/blob/vi-attr-init/src/attr/_make.py#L910

The variable name we used for the retval doesn't affect things (the contents of `locs` would have already fixed `__name__` etc).